### ISSUE TITLE
Build a success, execution startup a fail

### ISSUE THREAD

**Issue by JackDandy** [Chocobo1/qbittorent_msys2#3]
Hi,

Title says it all, the build was a success, however, when I copied the newly built `qbittorrent.exe` to my existing qB install folder, it failed to run... dialogs appear asking for different libraries, so I copied them over one by one as each error dialog requested.
However, I eventually hit a wall with <img src="https://vgy.me/JNY4ff.png">

I then followed the readme and step 4: Install using `pacman ...`, thinking all the deps are installed in the msys2 env, ran .. `I:\qB\msys64\mingw64\bin\qbittorrent.exe` but that too failed with the same error as the above image. 

Can you show how to statically build the dependencies into the exe - hopefully this would prevent this issue.

Thanks.

**Comment by Chocobo1** [comment]
Shoot, I did ran the built exe and didn't have the error...
Since I cannot reproduce, I guess I'll have to google it, this one looks similar: https://stackoverflow.com/a/24192521

**Comment by Chocobo1** [comment]
>Can you show how to statically build the dependencies into the exe - hopefully this would prevent this issue.

https://stackoverflow.com/questions/37524839/msys2-statically-link-output-binary

**Comment by JackDandy** [comment]
Hi,

I followed the step of creating sub folder `platforms` and copying into it `qwindows.dll`, with the other deps I already manually copied into place, qB at least starts up.

However, icon images are missing, and the window controls (e.g. buttons, lists, column headers) are winXPesque instead of _somewhat modern_ ...
<img src="https://vgy.me/xvLNHz.png">
  
compared to the expected...
<img src="https://vgy.me/DgkMOk.png">

Given your devel system has matured over time, all these issues are only present to someone like me, using this builder on a fresh system, therefore, can I suggest that you set up a VM and try this for yourself.  
  
I'm not confident that I can get this to statically compile, heck, even linked is proving non-trivial at the moment, you have the experience with qB, and you know what to look out for, given all these things are maybe something to be addressed in your PKGBUILD file.

Thanks.

**edit**: controls are restored to _vista style_ by adding /styles/qwindowsvistastyle.dll

**Comment by Chocobo1** [comment]
(wild guess) maybe qt5svg.dll is missing?

**Comment by Chocobo1** [comment]
I don't usually distribute dynamic linking exes, however if you're willing to share your findings/steps, I can add a link to it in readme (either a blog post or just here).

**Comment by JackDandy** [comment]
> I don't usually distribute dynamic linking exes,

Absolutely, that's why I was asking for static build steps, I've been sharing what I can as I "blindly" work through what you had available (amounting to what is linked build), but it would be great if you can complete this PKGBUILD to statically build the entire exe the way qB releases are normally. This way the troubles I have found would not be there for others coming along.

Also, the list i compiled was based on startup errors, if the guess of `qt5svg.dll` demonstrates anything, is that I have no idea the full range of deps, nor does the full range error on startup, this means i have likely missed something as I am not familiar.  Therefore, it is really down to you to complete a static packager :)

[edit]: still no imagery with that `qt5svg.dll` :(

**Comment by Chocobo1** [comment]
I don't think building statically linked exe can be achieved by changing only this PKGBUILD, as you see, every dependent package has to be modified & rebuilt (which is maintained in msys2 & mingw repo, not by me).
Pardon, I don't think it is feasible and I'm not really up to this challenge.

**Comment by JackDandy** [comment]
> every dependent package has to be modified & rebuilt

Actually, nope. 
**Every** package is ready built as `.dll` and also in the `\msys64\mingw64\lib` folder as `*.a` archives that each contain respective object link files, e.g. `libQt5Core.dll.a`. The `dll` versions are currently being linked, therefore, it is a case of finding out how to switch over and use their corresponding object files in your PKGBUILD for static building.

**Comment by rousselgeoffrey** [comment]
I successfully built the dev version with MSYS2 using your instructions Chocobo1.
(But i failed at building the stable version though, on some casting errors)
I just got into 1 issue at linking with a wrongly defined linker flag '-lIconv::Iconv', 
if you run into this issue:
-Open the file: "src/qBittorrent/conf.pri.in"
-Add theses 2 lines under "LIBS += $$EXTERNAL_LIBS":
```
LIBS -= -lIconv::Iconv
LIBS += -lIconv

```
And rebuild the package with: "makepkg --skippgpcheck --noconfirm --skipinteg --skipchecksums --nocheck --nodeps --noextract --holdver"

Now to use the binary out of MSYS2, i extracted the binary from the pkg to a separate folder,
I used ldd on the binary to copy the main dlls needed but i got into the same problem 
as JackDandy with missing icons, so i traced the binary with strace and copied the other dlls.
Copying theses dlls in the qbittorrent.exe directory solves the missing icons:
```

cp /mingw64/bin/libgcc_s_seh-1.dll .
cp /mingw64/bin/libwinpthread-1.dll .
cp /mingw64/bin/libstdc++-6.dll .
cp /mingw64/bin/libcrypto-1_1-x64.dll .
cp /mingw64/bin/Qt5Core.dll .
cp /mingw64/bin/libdouble-conversion.dll .
cp /mingw64/bin/libicuin64.dll .
cp /mingw64/bin/libicuuc64.dll .
cp /mingw64/bin/libicudt64.dll .
cp /mingw64/bin/libpcre2-16-0.dll .
cp /mingw64/bin/zlib1.dll .
cp /mingw64/bin/Qt5Gui.dll .
cp /mingw64/bin/libharfbuzz-0.dll .
cp /mingw64/bin/libfreetype-6.dll .
cp /mingw64/bin/libbz2-1.dll .
cp /mingw64/bin/libpng16-16.dll .
cp /mingw64/bin/libglib-2.0-0.dll .
cp /mingw64/bin/libintl-8.dll .
cp /mingw64/bin/libiconv-2.dll .
cp /mingw64/bin/libpcre-1.dll .
cp /mingw64/bin/libgraphite2.dll .
cp /mingw64/bin/Qt5Network.dll .
cp /mingw64/bin/Qt5Widgets.dll .
cp /mingw64/bin/Qt5WinExtras.dll .
cp /mingw64/bin/Qt5Xml.dll .
cp /mingw64/bin/libtorrent-rasterbar.dll .
cp /mingw64/bin/libssl-1_1-x64.dll .

cp 'C:\msys64\mingw64\share\qt5\plugins\platforms\qwindows.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\styles\qwindowsvistastyle.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\bearer\qgenericbearer.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qgif.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qicns.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qico.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qjp2.dll' .
cp 'C:\msys64\mingw64\bin\libjasper-4.dll' .
cp 'C:\msys64\mingw64\bin\libjpeg-8.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qjpeg.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qsvg.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qtga.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qtiff.dll' .
cp 'C:\msys64\mingw64\bin\libtiff-5.dll' .
cp 'C:\msys64\mingw64\bin\liblzma-5.dll' .
cp 'C:\msys64\mingw64\bin\libzstd.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qwbmp.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\imageformats\qwebp.dll' .
cp 'C:\msys64\mingw64\bin\libwebp-7.dll' .
cp 'C:\msys64\mingw64\bin\libwebpdemux-2.dll' .
cp 'C:\msys64\mingw64\share\qt5\plugins\iconengines\qsvgicon.dll' .

```
[EDIT]

Ok, so it works on my Windows7 but in order so that it works also on all systems and on Wine(Linux)
There is the need to preserve the Qt directories names in the qbittorrent.exe folder, that's the folders i had to copy: (with the previous named dlls in their respectives directories)

Directories:

```
iconengines/
imageformats/
platforms/
styles/

```

Their original locations is visible in the previous dlls i pasted, 
example: 'C:\msys64\mingw64\share\qt5\plugins\imageformats' for 'imageformats/'

If you run into an issue with a message like this:
"Application failed to start because it could not find or load the QT platform plugin"
You can try to create a file named "qt.conf" in the qbittorrent.exe folder with this content:
```
[Paths]
Plugins = .
```
But it works without this file here.

If you wonder why i get around Wine here, it is because i have a setup where i can run the same Qbittorrent binary on my Windows 10 or on Wine on Linux, i use it in portable mode, so i can seed the same torrents from an external drive on both systems without changing anything (got some virtual drives and i use the same drives letters on Windows and Linux(virtual drives))
